Userland/Utilities/hexdump.cpp

```diff
@@ -68,13 +71,29 @@ ErrorOr<int> serenity_main(Main::Arguments args)
     Span<u8> previous_line;
     static_assert(LINE_LENGTH_BYTES * 2 <= contents.size(), "Buffer is too small?!");
     size_t contents_size = 0;
+    size_t bytes_read = 0;
+    size_t bytes_remaining = 0;
+    int bytes_to_read = 0;
 
     int nread;
     auto state = State::Print;
-    while (true) {
-        nread = file->read(&contents[contents_size], BUFSIZ - contents_size);
+    bool is_input_remaining = true;
+    while (is_input_remaining) {
+        bytes_to_read = BUFSIZ - contents_size;
+
+        if (max_bytes.has_value()) {
+            bytes_remaining = max_bytes.value() - bytes_read;
```

Comment:
```suggestion
            bytes_remaining = max_bytes.value() - contents_size;
```

If we use `contents_size` here, then as far as I can tell we can get rid of the extra `bytes_read` variable entirely, because they measure the same thing. (Maybe rename `contents_size` -> `bytes_read`, since the name is confusing?)

Reply:
These variables are not interchangeable; one measures the total number of bytes read across the whole program, and the other measures the bytes read in the current buffer (with a maximum value of 1024). I've updated the names to hopefully alleviate the confusion.

Reply:
Oops, you're right!